internal/exec/stages/disks/luks.go

```diff
@@ -115,7 +115,18 @@ func (s *stage) createLuks(config types.Config) error {
 		}
 		keyFilePath := filepath.Join(distro.LuksInitramfsKeyFilePath(), luks.Name)
 		devAlias := execUtil.DeviceAlias(*luks.Device)
-		if util.NilOrEmpty(luks.KeyFile.Source) {
+		sources, err := luks.KeyFile.GetSources()
+		if err != nil {
+			return err
+		}
+		src := ""
+		for i := 0; i < len(sources); i++ {
+			if sources[i] != "" {
+				src = string(sources[i])
+				break
+			}
+		}
+		if util.NilOrEmpty(&src) {
```

Comment:
I wonder if we shouldn't just create a new utility function that checks the output of `GetSources` to check if the only source is `""`.

Reply:
i tried, but than end up with recursive dependency ( because of `[] Source` ).